teuthology/task/install/rpm.py

```diff
@@ -132,12 +144,12 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
     repos_only = config.get('repos_only')
 
     if repos:
-        log.debug("Adding repos: %s" % repos)
         if dist_release in ['opensuse', 'sle']:
-            _zypper_addrepo(remote, repos)
+            _zypper_wipe_existing_repos(remote)
         else:
             raise Exception('Custom repos were specified for %s ' % remote_os +
-                            'but these are currently not supported')
+                            'but these are currently not supported on that OS')
+        _zypper_addrepo(remote, repos)
```

Comment:
Why this line has been moved here, out of 'opensuse', 'sle' scope? It will fail for other distro.